Sources/MapboxCoreNavigation/BillingHandler.swift

```diff
@@ -307,6 +307,34 @@ final class BillingHandler {
         }
     }
 
+    /**
+     Starts a new billing session in `billingService` if a session with `uuid` exists and active.
+
+     Use this method to force `billingService` to start a new billing session. 
+     */
+    func beginNewBillingSessionIfRunning(with uuid: UUID) {
+        lock.lock()
+
+        guard let session = _sessions[uuid], !session.isPaused else {
+            return
+        }
+
+        lock.unlock()
+
+        billingService.beginBillingSession(for: session.type) { [logger] error in
+            os_log("New trip session isn't started. Please check that you have the correct Mapboox Access Token",
```

Comment:
If we’re pretty sure this is misconfiguration on the part of the developer, rather than internal inconsistency out of the developer’s control, then this should be a precondition failure to get the developer’s attention, similar to what we do when no access token has been set:

https://github.com/mapbox/mapbox-navigation-ios/blob/6b827eeb5009f0fe6ddf6db7fc61494232180d82/Sources/MapboxCoreNavigation/BillingHandler.swift#L184

Reply:
`tokenValidationFailed` can be called when server doesn't validate token or token is empty (we already have check for that). If we put precondition here we might crash developers' apps due to bug on server. It is far from desirable. I can try to put assert here though.